packages/devextreme/js/__internal/ui/collection/m_search_box_mixin.ts

```diff
@@ -1,178 +1,154 @@
-    const rootElementClassName = this._addWidgetPrefix('with-search');
+export type SearchBoxControllerOptions = SearchBoxMixinOptions & {
+  tabIndex?: number;
+  onValueChanged?: (value: string) => void;
+};
 
-    if (!searchEnabled) {
-      $element.removeClass(rootElementClassName);
-      this._removeSearchBox();
-      return;
-    }
+interface SearchBoxControllerProps {
+  createEditor: (
+    $element: dxElementWrapper,
+    component: TextBox,
+    options: Record<string, unknown>,
+  ) => TextBox;
+  widgetPrefix: string;
+}
 
-    const editorOptions = this._getSearchEditorOptions();
+class SearchBoxController {
+  _createEditor: SearchBoxControllerProps['createEditor'];
 
-    if (this._searchEditor) {
-      this._searchEditor.option(editorOptions);
-    } else {
-      $element.addClass(rootElementClassName);
-      this._$searchEditorElement = $('<div>').addClass(searchBoxClassName).prependTo($element);
-      this._searchEditor = this._createComponent(this._$searchEditorElement, EditorClass, editorOptions);
-    }
-  },
+  _widgetPrefix: string;
 
-  _removeSearchBox(): void {
-    this._$searchEditorElement && this._$searchEditorElement.remove();
-    delete this._$searchEditorElement;
-    delete this._searchEditor;
-  },
+  _$element!: dxElementWrapper | null;
 
-  _getSearchEditorOptions() {
-    const that = this;
-    const userEditorOptions = that.option('searchEditorOptions');
-    const searchText = messageLocalization.format('Search');
+  _editor!: TextBox | null;
 
-    return extend({
-      mode: 'search',
-      placeholder: searchText,
-      tabIndex: that.option('tabIndex'),
-      value: that.option('searchValue'),
-      valueChangeEvent: 'input',
-      inputAttr: {
-        'aria-label': searchText,
-      },
-      onValueChanged(e) {
-        const searchTimeout = that.option('searchTimeout');
-        that._valueChangeDeferred = Deferred();
-        clearTimeout(that._valueChangeTimeout);
-
-        that._valueChangeDeferred.done(function () {
-          this.option('searchValue', e.value);
-        }.bind(that));
-
-        if (e.event && e.event.type === 'input' && searchTimeout) {
-          that._valueChangeTimeout = setTimeout(() => {
-            that._valueChangeDeferred.resolve();
-          }, searchTimeout);
-        } else {
-          that._valueChangeDeferred.resolve();
-        }
-      },
-    }, userEditorOptions);
-  },
+  _valueChangeDeferred!: DeferredObj<unknown>;
 
-  _getAriaTarget() {
-    if (this.option('searchEnabled')) {
-      return this._itemContainer(true);
-    }
-    return this.callBase();
-  },
+  // eslint-disable-next-line no-restricted-globals
+  _valueChangeTimeout!: ReturnType<typeof setTimeout>;
 
-  _focusTarget() {
-    if (this.option('searchEnabled')) {
-      return this._itemContainer(true);
-    }
+  _onSearchBoxValueChanged?: (value: string) => void;
 
-    return this.callBase();
-  },
+  static EditorClass: any = stubComponent('TextBox');
```

Comment:
We need add types for stubComponent() function and get rid of any type because of only we use this function